internal/distro/distro.go

```diff
@@ -82,7 +82,7 @@ type ImageType interface {
 	// Returns an osbuild manifest, containing the sources and pipeline necessary
 	// to build an image, given output format with all packages and customizations
 	// specified in the given blueprint.
-	Manifest(b *blueprint.Customizations, options ImageOptions, repos []rpmmd.RepoConfig, packageSpecs, buildPackageSpecs []rpmmd.PackageSpec, seed int64) (Manifest, error)
+	Manifest(b *blueprint.Customizations, options ImageOptions, repos []rpmmd.RepoConfig, packageSpecSets [][]rpmmd.PackageSpec, seed int64) (Manifest, error)
```

Comment:
I wonder if a `map[string][]rpmmd.PackageSpec` is preferable here, with string like "build" and "base". Not sure if strings as indices are less error-prone, but it might make it easier to debug?

Reply:
I agree.  My current version of the open PR (not pushed yet) uses a map and it's much cleaner.  The names/keys don't need to have any real meaning (e.g., they don't need to match pipeline or stage names), but it's easier to work with and more readable.

Reply:
good point, will change

Reply:
Done.